instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java

```diff
@@ -232,10 +232,11 @@ public void end(
 
     if (!requestListeners.isEmpty() || !requestMetricListeners.isEmpty()) {
       long endNanos = getNanos(endTime);
-      for (RequestListener requestListener : requestListeners) {
+      // TODO (trask) call end in the reverse order that start was called?
```

Comment:
We probably don't have to do that - unlike `start()`, the `end()` method cannot modify the context; also, we're passing the timestamp as an argument, so the exact time of invocation should not matter too.

Reply:
ya, I just always think of start/end pairs telescoping :)